erdjs/src/dapp/walletConnectProvider.ts

```diff
@@ -0,0 +1,198 @@
+import WalletClient from "@walletconnect/client";
+import { IProvider } from "../interface";
+import { Transaction } from "../transaction";
+import { Address } from "../address";
+import { IDappProvider } from "./interface";
+import { Signature } from "../signature";
+
+export class WalletConnectProvider extends EventTarget implements IDappProvider {
+    provider: IProvider;
+    walletConnectBridge: string;
+    address: string = "";
+    walletConnector: WalletClient | undefined;
+    private onWalletConnectLogin: Event = new Event("onWalletConnectLogin");
+    private onWalletConnectDisconect: Event = new Event("onWalletConnectDisconect");
+    constructor(httpProvider: IProvider, walletConnectBridge: string = "") {
+        super();
+        this.provider = httpProvider;
+        this.walletConnectBridge = walletConnectBridge;
+    }
+
+    /**
+     * Initiates wallet connect client.
+     */
+    async init(): Promise<boolean> {
+        this.walletConnector = new WalletClient({
+            bridge: this.walletConnectBridge,
+        });
+        this.walletConnector.on("connect", this.onConnect);
+        this.walletConnector.on("session_update", this.onDisconnect);
+        this.walletConnector.on("disconnect", this.onDisconnect);
+        return true;
+    }
+
+    /**
+     * Returns true if init() was previously called successfully
+     */
+    isInitialized(): boolean {
+        return !!this.walletConnector;
+    }
+
+    /**
+     * Mocked function, returns isInitialized as an async function
+     */
+    isConnected(): Promise<boolean> {
+        return new Promise((resolve, _) => resolve(this.isInitialized()));
+    }
+
+    /**
+     *
+     */
+    async login(): Promise<string> {
+        const chainId = 508;
```

Comment:
Perhaps move this to a constant somewhere in walletcore package?

I see here we have the derivation path:
https://github.com/ElrondNetwork/elrond-sdk/blob/development/erdjs/src/walletcore/mnemonic.ts#L7

Reply:
done, moved